examples/timebase_master.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+#  timebase_master.py
+#
+"""A simple JACK timebase master."""
+
+import argparse
+import sys
+from threading import Event
+
+import jack
+
+
+class TimebaseMasterClient(jack.Client):
+    def __init__(self, name, bpm=120.0, meter='4/4', beats_per_bar=4,
+                 beat_type=4, ticks_per_beat=1920, conditional=False,
+                 debug=False, **kw):
+        super().__init__(name, **kw)
+        self.beats_per_bar = beats_per_bar
+        self.beat_type = beat_type
+        self.bpm = bpm
+        self.conditional = conditional
+        self.debug = debug
+        self.ticks_per_beat = int(ticks_per_beat)
+        self.stop_event = Event()
+        self.set_shutdown_callback(self.shutdown)
+
+    def shutdown(self, status, reason):
```

Comment:
Since `TimebaseMasterClient.stop_event` is public, I figured `shutdown` should be too. Agreed, encapsulation could be better, e.g. by making `stop_event` private and adding a method (e.g. `run()`) that does the whole business of reseting the event, setting the shutdown and timebase callbacks and then waiting for the event.